proposals/NNNN-resource-attributes-in-dxil-and-spirv.md

```diff
@@ -0,0 +1,386 @@
+sections awkwardly early in the compiler.
+
+[SPV_EXT_fragment_shader_interlock]:
+    https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/EXT/SPV_EXT_fragment_shader_interlock.asciidoc
+
+|                                    |
+|------------------------------------|
+| RasterizerOrderedTexture1D         |
+| RasterizerOrderedTexture1DArray    |
+| RasterizerOrderedTexture2D         |
+| RasterizerOrderedTexture2DArray    |
+| RasterizerOrderedTexture3D         |
+| RasterizerOrderedBuffer            |
+| RasterizerOrderedByteAddressBuffer |
+| RasterizerOrderedStructuredBuffer  |
+
+Feedback textures aren't implemented in DXC.
+
+|                                    |
+|------------------------------------|
+| FeedbackTexture2D                  |
+| FeedbackTexture2DArray             |
+
+### Attributes
+
+From all of this, we're able to come up with a set of attributes that can
+represent all of the texture and buffer types. These should carry sufficient
+information to lower to both of the targets:
+
+|                                    | RC   | Type   | ROV | Dim  | MS  | Feedback | Array | Raw | Row |
+|------------------------------------|------|--------|-----|------|-----|----------|-------|-----|-----|
+| Texture1D                          | SRV  | vec4   | -   | 1D   | -   | -        | -     | -   | -   |
+| Texture1DArray                     | SRV  | vec4   | -   | 1D   | -   | -        | yes   | -   | -   |
+| Texture2D                          | SRV  | vec4   | -   | 2D   | -   | -        | -     | -   | -   |
+| Texture2DArray                     | SRV  | vec4   | -   | 2D   | -   | -        | yes   | -   | -   |
+| Texture2DMS                        | SRV  | vec4   | -   | 2D   | yes | -        | -     | -   | -   |
+| Texture2DMSArray                   | SRV  | vec4   | -   | 2D   | yes | -        | yes   | -   | -   |
+| Texture3D                          | SRV  | vec4   | -   | 3D   | -   | -        | -     | -   | -   |
+| TextureCUBE                        | SRV  | vec4   | -   | Cube | -   | -        | -     | -   | -   |
+| TextureCUBEArray                   | SRV  | vec4   | -   | Cube | -   | -        | yes   | -   | -   |
+| RWTexture1D                        | UAV  | vec4   | -   | 1D   | -   | -        | -     | -   | -   |
+| RWTexture1DArray                   | UAV  | vec4   | -   | 1D   | -   | -        | yes   | -   | -   |
+| RWTexture2D                        | UAV  | vec4   | -   | 2D   | -   | -        | -     | -   | -   |
+| RWTexture2DArray                   | UAV  | vec4   | -   | 2D   | -   | -        | yes   | -   | -   |
+| RWTexture2DMS                      | UAV  | vec4   | -   | 2D   | yes | -        | -     | -   | -   |
+| RWTexture2DMSArray                 | UAV  | vec4   | -   | 2D   | yes | -        | yes   | -   | -   |
+| RWTexture3D                        | UAV  | vec4   | -   | 3D   | -   | -        | -     | -   | -   |
```

Comment:
These look good. The individual attributes map nicely to individual operands to spirv.Image.